packages/office-ui-fabric-react/src/components/Nav/Nav.tsx

```diff
@@ -82,39 +84,39 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
       (this._hasExpandButton ? _indentWithExpandButton : _indentNoExpandButton)).toString(10) + 'px';
 
     return (
-        <a
-          className={ css('ms-Nav-link') }
-          style={ { [isRtl ? 'paddingRight' : 'paddingLeft'] : paddingBefore } }
-          href={ link.url || 'javascript:' }
-          onClick={ this._onNavAnchorLinkClicked.bind(this, link) }
-          aria-label={ link.ariaLabel }
-          title={ link.title || link.name }
-          target={ link.target }
+      <a
+        className={ css('ms-Nav-link') }
+        style={ { [isRtl ? 'paddingRight' : 'paddingLeft']: paddingBefore } }
+        href={ link.url || 'javascript:' }
+        onClick={ this._onNavAnchorLinkClicked.bind(this, link) }
+        aria-label={ link.ariaLabel }
+        title={ link.title || link.name }
+        target={ link.target }
         >
-         { link.iconClassName && <i className={ css('ms-Icon', 'ms-Nav-IconLink', link.iconClassName) }></i> }
-         { this.props.onRenderLink(link)}
-        </a>
+        { link.iconClassName && <i className={ css('ms-Icon', 'ms-Nav-IconLink', link.iconClassName) }></i> }
+        { this.props.onRenderLink(link) }
+      </a>
     );
   }
 
   private _renderButtonLink(link: INavLink, linkIndex: number) {
     return (
       <Button
         className={ css('ms-Nav-link ms-Nav-linkButton', { 'isOnExpanded': this._hasExpandButton }) }
-          buttonType={ ButtonType.command }
-          icon={ link.icon }
-          description={ link.title || link.name }
-          onClick={ this._onNavButtonLinkClicked.bind(this, link) }>
-          { link.name }
+        buttonType={ ButtonType.command }
+        icon={ link.icon }
+        description={ link.title || link.name }
+        onClick={ this._onNavButtonLinkClicked.bind(this, link) }>
+        { link.name }
       </Button>);
   }
 
   private _renderCompositeLink(link: INavLink, linkIndex: number, nestingLevel: number): React.ReactElement<{}> {
-    const isLinkSelected: boolean = _isLinkSelected(link, this.state.selectedKey);
+    const isLinkSelected: boolean = _isLinkSelected(link, this.state.selectedKey, this.props.selectedKey);
 
     return (
       <div key={ link.key || linkIndex }
-           className={ css('ms-Nav-compositeLink', { ' is-expanded': link.isExpanded, 'is-selected': isLinkSelected }) }>
+        className={ css('ms-Nav-compositeLink', { ' is-expanded': link.isExpanded, 'is-selected': isLinkSelected }) }>
         { (nestingLevel === 0 && link.links && link.links.length > 0 ?
           <button
```

Comment:
Button needs a label. It should also have aria-expanded for the region which expands/collapses